dhis-2/dhis-support/dhis-support-jdbc/src/main/java/org/hisp/dhis/jdbc/statementbuilder/AbstractStatementBuilder.java

```diff
@@ -450,9 +454,10 @@ public String getBoundaryCondition( ProgramIndicator programIndicator, Date repo
 
     @Override
     public String getBoundaryCondition( AnalyticsPeriodBoundary boundary, ProgramIndicator programIndicator,
-        Date reportingStartDate, Date reportingEndDate )
+        String timeField, Date reportingStartDate, Date reportingEndDate )
     {
-        String column = boundary.isEventDateBoundary() ? AnalyticsPeriodBoundary.DB_EVENT_DATE
+        String column = boundary.isEventDateBoundary()
+            ? Optional.ofNullable( timeField ).orElse( AnalyticsPeriodBoundary.DB_EVENT_DATE )
```

Comment:
If we are supporting enrollment type indicators as well this needs a tweak.

Reply:
> I like the restructuring you did. In the user story the description also included "events entered/TEIs created/Enrollments created" - I would guess we should support enrollment analytics as well in this iteration? Now you have dived down into it might be the best time to add the support also for analytics. This is really a question/decision for Scott to make, but if the decision is not to support it for enrollments, we should perhaps instead return a validation message telling the user that lastUpdated/timefields are not supported for enrollments.

as discussed, I added a specific jira ticket for enrollments